docs/src/main/java/io/quarkus/docs/generation/AllConfigGenerator.java

```diff
@@ -0,0 +1,149 @@
+package io.quarkus.docs.generation;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.resolution.ArtifactRequest;
+import org.eclipse.aether.resolution.ArtifactResult;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.quarkus.annotation.processor.generate_doc.ConfigDocItem;
+import io.quarkus.annotation.processor.generate_doc.ConfigDocItemScanner;
+import io.quarkus.annotation.processor.generate_doc.ConfigDocSection;
+import io.quarkus.annotation.processor.generate_doc.ConfigDocWriter;
+import io.quarkus.bootstrap.resolver.AppModelResolverException;
+import io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver;
+import io.quarkus.docs.generation.ExtensionJson.Extension;
+
+public class AllConfigGenerator {
+    public static void main(String[] args)
+            throws AppModelResolverException, JsonParseException, JsonMappingException, IOException {
+        MavenArtifactResolver resolver = MavenArtifactResolver.builder().build();
+        // This is where we produce the entire list of extensions
+        String jsonPath = "devtools/core-extensions-json/target/extensions.json";
+        ObjectMapper mapper = new ObjectMapper();
+
+        // let's read it (and ignore the fields we don't need)
+        ExtensionJson extensionJson = mapper.readValue(new File(jsonPath), ExtensionJson.class);
+        
+        // now get all the listed extension jars via Maven
+        List<ArtifactRequest> requests = new ArrayList<>(extensionJson.extensions.size());
+        Map<String, Extension> extensionsByGav = new HashMap<>();
+        Map<String, Extension> extensionsByConfigRoots = new HashMap<>();
+        for (Extension extension : extensionJson.extensions) {
+            ArtifactRequest request = new ArtifactRequest();
+            Artifact artifact = new DefaultArtifact(extension.groupId, extension.artifactId, "jar", "999-SNAPSHOT");
```

Comment:
I'm fixing the missing version as well.

Reply:
Yeah, I was thinking of passing that version as argument to the program, which I get from the `pom.xml`.

Reply:
I'm now taking it as parameter from the pom.